core-bundle/src/Routing/PageUrlGenerator.php

```diff
@@ -38,6 +38,9 @@ public function __construct(RouteProviderInterface $provider, PageRegistry $page
         $this->pageRegistry = $pageRegistry;
     }
 
+    /**
+     * @throws RouteParametersException
```

Comment:
We are not using `@throws` annotations anywhere. Why does PHPStan complain about this one?

Reply:
I'm not sure why PHPStan cannot figure this out, but IMHO it's correct to annotate exceptions that are thrown as apart of the "business logic", i.e. no guards but custom exceptions you would catch as part of the normal control flow.

Reply:
What is the exact PHPStan error message?

Reply:
See https://github.com/contao/contao/pull/3664#issuecomment-965826213 or click the link in the initial description.

Reply:
> Dead catch - Contao\CoreBundle\Exception\RouteParametersException is never thrown in the try

Reply:
Then it is obviously not a Contao issue. We should probably report it and ignore it in the PHPStan config file until it has been fixed.

Reply:
I spent some time debugging it. It is indeed no bug in PHPStan. The reason the error is reported is because the `Symfony\Component\Routing\Generator\UrlGeneratorInterface` annotates the following exceptions that may be thrown (regular control flow):

https://github.com/symfony/symfony/blob/b560ec4a1d11f7b7e78a4bb9ed75b84397222a26/src/Symfony/Component/Routing/Generator/UrlGeneratorInterface.php#L80-L82

Our own `RouteParametersException` is of course none of them. IMHO this is a bug because now we're not following the interface anymore. People implementing it are only supposed to handle the annotated exceptions.

I made the `RouteParametersException` extend from `InvalidParameterException` in 6158ad1115f9eb85af47cce69671e1ce7a772a20. This way our `PrettyErrorScreenListener` still can access the the additional route information but we're compliant again. 
@aschempp Is this the right exception to extend?